src/clients/Client.cc

```diff
@@ -507,9 +507,7 @@ Client::haveParsedReplyHeaders()
     maybePurgeOthers();
 
     // adaptation may overwrite old offset computed using the virgin response
-    const bool partial = theFinalReply->content_range &&
-                         theFinalReply->sline.status() == Http::scPartialContent;
-    currentOffset = partial ? theFinalReply->content_range->spec.offset : 0;
+    currentOffset = theFinalReply->contentRange() ? theFinalReply->contentRange()->spec.offset : 0;
```

Comment:
Please keep the original `partial` constant. It both documents the logic behind the code and exposes a possible problem in that logic (some partial responses do not satisfy the `partial` initialization condition!). Addressing that concern is outside this issue scope, but hiding it is not a good idea.

Reply:
Done.